### DISCUSSION TITLE
TRVs with air to water heat pump

### DISCUSSION THREAD

**Discussion by fredericseiler** [jmcollin78/versatile_thermostat#363] [Q&A]
Hi all,

I have this situation where I want to control my central heating system (air to water heat pump which has its own room/ambient thermostat in the lobby) with the status of my vtherms (TRV on water radiator in each room).

I'm looking for advices on the best strategy to adopt. Here's my initial plan:

When all TRVs become inactive (valves at 0%)
- force heat pump target temp to heat pump current temp (or less?)
- (optional) stop heat pump if started

Whenever a TRV needs heating (any valve > 0%)
- new target temp = heat pump current temp + (valve target temp - valve current temp)
- if new target temp > heat pump *actual* target temp
  - force heat pump target temp to new target temp
  - start heat pump if stopped

Then wait for valve or temperature change and reroll (with cooldown ?)

Note: the stop step is optional because of the inverter technology used in heat pumps; I'm not quite sure yet if it's a smart thing to do.

What is bugging me is that the TPI algorithm will put a TRV in heating mode while the target temp is less than the current temp because of the use of outside sensor, so I can't just apply the difference to the heat pump current temperature.

Any thoughts?

**Comment by jmcollin78** [comment]
A preliminary question please : can you control the open percent of the TRV directly (Shelly TRV does that for example) ?
The question of "how HA will control your TRV is important" in the response. If you cannot control open percent directly, you will have to create `over_climate` VTherm and then there is no TPI algorithm involved.

**Comment by fredericseiler** [comment]
Sorry, lack of details:

The TRVs are all Shellys (and working great in HA thanks to your amazing integration)
The heat pump is also fully controlled in HA (via Airzone Aidoo's local API and Noltari's great work)

As for the technical implementation, it doesn't matter (yet); HA automations, node-red, HA api with php/python/you name it, the job will be done.

The real problem is: with a typical water boiler scenario (like fuel, gas, or electric), it's only a boolean issue (on or off), like you did recently with the central boiler control, but with a heat pump, it's a whole different situation, I want to/have to take into consideration the inverter tech, and I can't figure out the best way to handle it (and I'm pretty sure the answer is much simpler than I think).

I haven't found anyone on the web trying this sort of setup (heat pump with smart TRVs), am I missing something ?

**Reply by jmcollin78** [reply]
This is interesting and I would like to understand more waht you call "take into consideration the inverter tech". If you have a discord account or an account on forum.hacf.fr I would to have explainations in French. In the forum.hacf.fr, you can contact me on my MP. Else on discord my id is jm.collin.

**Comment by jmcollin78** [comment]
@fredericseiler and for others which may fall into this, it is possible with the 'central boiler control' feature to adjust the target temp and not only to start / stop.

The service (in HA terms) you call for turning on or off is fully configurable and can be configured to change the target temperature of the boiler.

This is fully explain here: https://github.com/jmcollin78/versatile_thermostat/blob/main/README.md#control-of-a-central-boiler with some examples.
